ddpui/api/transform_api.py

```
@@ -689,78 +690,141 @@ def get_warehouse_datatypes(request):
+    try:
+        lock: CanvasLock = orgdbt.canvas_lock
+        if lock.is_expired():
+            raise HttpError(410, "Lock has expired")
+        if lock.locked_by != orguser:
+            raise HttpError(403, "You can only refresh your own locks")
+
+        # Refresh lock with 2-minute duration
+        lock.expires_at = timezone.now() + timedelta(minutes=2)
+        lock.save()
+
+        logger.info(f"Refreshed lock for canvas")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)